diff --git a/libstdc++-v3/include/Makefile.in b/libstdc++-v3/include/Makefile.in
index 9fd4ab4848c..5f0d67a2d0b 100644
--- a/libstdc++-v3/include/Makefile.in
+++ b/libstdc++-v3/include/Makefile.in
@@ -380,6 +380,12 @@ WARN_CXXFLAGS = \
 # -I/-D flags to pass when compiling.
 AM_CPPFLAGS = $(GLIBCXX_INCLUDES) $(CPPFLAGS)
 
+# EQT special header helpers
+__eqt_srcdir = ${glibcxx_srcdir}/include/__eqt
+__eqt_builddir = ./__eqt
+__eqt_headers = \
+	${__eqt_srcdir}/excstring.hpp
+
 # Standard C++ includes.
 std_srcdir = ${glibcxx_srcdir}/include/std
 std_builddir = .
@@ -1440,7 +1446,7 @@ PCHFLAGS = -x c++-header -nostdinc++ $(CXXFLAGS) $(VTV_PCH_CXXFLAGS)
 # List of all timestamp files.  By keeping only one copy of this list, both
 # CLEANFILES and all-local are kept up-to-date.
 allstamped = \
-	stamp-std stamp-bits stamp-bits-sup stamp-pstl stamp-c_base stamp-c_compatibility \
+	stamp-__eqt stamp-std stamp-bits stamp-bits-sup stamp-pstl stamp-c_base stamp-c_compatibility \
 	stamp-backward stamp-ext stamp-pb stamp-tr1 stamp-tr2 stamp-decimal \
 	stamp-experimental stamp-experimental-bits stamp-debug stamp-parallel \
 	stamp-host
@@ -1632,6 +1638,11 @@ all-local: ${allstamped} ${allcreated}
 # systems with broken mkdir.  Call mkdir unconditionally because
 # it is just as cheap to avoid going through the shell.
 # Ignore errors from $(LN_S) because the links may already exist.
+stamp-__eqt: ${__eqt_headers}
+	@-mkdir -p ${__eqt_builddir}
+	@-cd ${__eqt_builddir} && $(LN_S) $? . 2>/dev/null
+	@$(STAMP) stamp-__eqt
+
 stamp-std: ${std_headers}
 	@-mkdir -p ${std_builddir}
 	@-cd ${std_builddir} && $(LN_S) $? . 2>/dev/null
@@ -1963,6 +1974,9 @@ update-version:
 install-data-local: install-headers
 install-headers:
 	$(mkinstalldirs) $(DESTDIR)${gxx_include_dir}
+	$(mkinstalldirs) $(DESTDIR)${gxx_include_dir}/${__eqt_builddir}
+	for file in ${__eqt_headers}; do \
+	  $(INSTALL_DATA) $${file} $(DESTDIR)${gxx_include_dir}/${__eqt_builddir}; done
 	$(mkinstalldirs) $(DESTDIR)${gxx_include_dir}/${bits_builddir}
 	for file in ${bits_headers}; do \
 	  $(INSTALL_DATA) $${file} $(DESTDIR)${gxx_include_dir}/${bits_builddir}; done
@@ -2056,9 +2070,10 @@ clean-local:
 # developer tries to create them via make in the include build
 # directory. (This is more of an example of how this kind of rule can
 # be made.)
-.PRECIOUS: $(std_headers) $(c_base_headers) $(tr1_headers) $(tr2_headers)
+.PRECIOUS: $(__eqt_headers) $(std_headers) $(c_base_headers) $(tr1_headers) $(tr2_headers)
 	   $(decimal_headers) $(ext_headers) $(experimental_headers)
 	   $(experimental_bits_headers)
+$(__eqt_headers): ; @:
 $(std_headers): ; @:
 $(c_base_headers): ; @:
 $(tr1_headers): ; @:
