diff --git a/gdb/configure b/gdb/configure
index 1531f62f76a..c799192ef0b 100755
--- a/gdb/configure
+++ b/gdb/configure
@@ -27005,7 +27005,7 @@ return waddstr ();
   return 0;
 }
 _ACEOF
-for ac_lib in '' ncursesw ncurses cursesX curses; do
+for ac_lib in '' ncurses cursesX curses; do
   if test -z "$ac_lib"; then
     ac_res="none required"
   else
@@ -27101,7 +27101,7 @@ return tgetent ();
   return 0;
 }
 _ACEOF
-for ac_lib in '' termcap tinfow tinfo curses ncursesw ncurses; do
+for ac_lib in '' termcap tinfo curses ncurses; do
   if test -z "$ac_lib"; then
     ac_res="none required"
   else
@@ -29246,7 +29246,6 @@ for ac_header in  \
   ncurses.h \
   ncurses/ncurses.h \
   ncurses/term.h \
-  ncursesw/ncurses.h \
 
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
